d3dcore/include/d3dcore/d3d_info.h

```diff
@@ -0,0 +1,253 @@
+#pragma once
+
+#define INITGUID
+#include "dxdiag.h"
+#include <Windows.h>
+
+#include "d3d_utils.h"
+
+#include <string>
+#include <memory>
+#include <assert.h>
+#include <iostream>
+#include <string>
+#include <any>
+
+
+//-----------------------------------------------------------------------------
+// Defines, and constants
+//-----------------------------------------------------------------------------
+#define SAFE_DELETE(p)       { if(p) { delete (p);     (p)=NULL; } }
+#define SAFE_DELETE_ARRAY(p) { if(p) { delete[] (p);   (p)=NULL; } }
+#define SAFE_RELEASE(p)      { if(p) { (p)->Release(); (p)=NULL; } }
+#define SAFE_BSTR_FREE(x)    if(x) { SysFreeString( x ); x = NULL; }
+#define EXPAND(x)            x, sizeof(x)/sizeof(TCHAR)
+
+
+
+namespace d3dcore {
+
+	/*enum class SDKVersion {
+	DX9: DXDIAG_DX9_SDK_VERSION,
+	DX11: 112,		
+	};*/
+
+	template<typename Container>
+	class D3DInfo
+	{
+	public:
+		D3DInfo();
```

Comment:
If you have empty c-tor, use = default() semantic. If you don't need copy-assignment, always explicitly forbid it with = delete!